File: patterns/primer-mit-hablladin/rules-js/primer-mit-hablladin-manual.js
const GLSM_OFF = "0";
const GLSM_ON = "1";

rules.JSRule({
  name: "Garage Lights Commanded On",
  triggers: [triggers.ItemStateChangeTrigger("GarageLightsProxy", undefined, "ON")],
  execute: (event) => {
    items.getItem("GLSM").postUpdate(GLSM_ON);
    console.log("GarageLightsProxy -> ON, GLSM -> " + GLSM_ON);
  }
});

rules.JSRule({
  name: "Garage Lights Commanded Off",
  triggers: [triggers.ItemStateChangeTrigger("GarageLightsProxy", undefined, "OFF")],
  execute: (event) => {
    items.getItem("GLSM").postUpdate(GLSM_OFF);
    console.log("GarageLightsProxy -> OFF, GLSM -> " + GLSM_OFF);
  }
});

File: patterns/primer-mit-hablladin/rules-python/primer_mit_hablladin_manual.py
from openhab import rule, Registry
from openhab.triggers import ItemStateChangeTrigger

GLSM_ON = "1"
GLSM_OFF = "0"


@rule(triggers=[ItemStateChangeTrigger("GarageLightsProxy", state="ON")])
class GarageLightsCommandedOn:
    def execute(self, module, input):
        Registry.getItem("GLSM").postUpdate(GLSM_ON)


@rule(triggers=[ItemStateChangeTrigger("GarageLightsProxy", state="OFF")])
class GarageLightsCommandedOff:
    def execute(self, module, input):
        Registry.getItem("GLSM").postUpdate(GLSM_OFF)
